src/db/driver.cr

```diff
@@ -0,0 +1,34 @@
+module DB
+  # Database driver implementors must subclass `Driver`,
+  # register with a driver_name using `DB#register_driver` and
+  # override the factory method `#build_connection`.
+  #
+  # ```
+  # require "db"
+  #
+  # class FakeDriver < Driver
+  #   def build_connection(db)
+  #     FakeConnection.new db
+  #   end
+  # end
+  #
+  # DB.register_driver "fake", FakeDriver
+  # ```
+  #
+  # Access to this fake datbase will be available with
+  #
+  # ```
+  # DB.open "fake://..." do |db|
+  #   # ... use db ...
+  # end
+  # ```
+  #
+  # Refer to `Connection`, `Statement` and `ResultSet` for further
+  # driver implementation instructions.
+  abstract class Driver
+    def initialize
```

Comment:
@asterite  I thought it will force subclass to always have a parameterless constructor that way, but https://play.crystal-lang.org/#/r/sor shows I was wrong. :-( `DB#build_database` assumes a parameter less constructor. But that will be easily depicted by the driver implementor during testing.